services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -212,12 +221,15 @@ public MongoDbBasedRegistrationService(
         Objects.requireNonNull(deviceKey);
         Objects.requireNonNull(span);
 
-        return findDeviceDocument(deviceKey.getTenantId(), deviceKey.getDeviceId())
+        return tenantExists(deviceKey.getTenantId(), span)
```

Comment:
I do not think that this is necessary because the assert Registration operation explicitly defines how to handle the case when there is no device matching the criteria ...